### DISCUSSION TITLE
File uploading UX

### DISCUSSION THREAD

**Discussion by AbdulrhmnGhanem** [kitspace/kitspace-v2#288] [Ideas]
I think the next step is to solidify the uploading process. What I think of now is to add a progress bar to indicate what is going on then redirect because there is an interval between drping files and redirecting to the project page. @kasbah, what do you think? Do you have other suggestions?

**Comment by kasbah** [comment]
Most of the work would be improving the API in the processor?

**Reply by AbdulrhmnGhanem** [reply]
No, I am talking about what happens between dropping files/folders and redirecting to the project page. Just this part for now. I guess all the changes will be in the frontend.

**Reply by kasbah** [reply]
Sounds reasonable, yeah. The way I have been thinking about it is having the project page load in a way, where the general scaffolding is loaded instantly and indvidual bits come in as and when they are ready. This would improve the general loading experience of the page, not just when importing.

**Reply by kasbah** [reply]
Though it depends how it's done. Could be an annoying experience as well with bits popping in every few seconds.
